packages/api/test/add-build-test.js

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const { describe, it, after, before } = require('mocha');
+const { Firestore } = require('@google-cloud/firestore');
+const assert = require('assert');
+
+const TestCaseRun = require('../lib/testrun');
+const addBuild = require('../src/add-build');
+
+const TESTING_COLLECTION_BASE = 'repositories-testsuite-';
+
+// The three builds that will be added
+const buildInfo = [
+  {
+    repoId: encodeURIComponent('nodejs/node'),
+    organization: 'nodejs',
+    buildId: '11111',
+    url: 'https://github.com/nodejs/WRONG', // URL starts off wrong
```

Comment:
good call mocking out the builds, this is exactly the approach I would have suggested 👍